openpype/lib/project_backpack.py

```diff
@@ -141,6 +141,16 @@ def pack_project(
     if not destination_dir:
         destination_dir = root_path
 
+    if not destination_dir:
+        if only_documents:
+            raise ValueError((
+                "Destination dir must be passed."
+                " Use '--dirpath {output dir path}' if using command line."
+            ))
```

Comment:
Should this error message also mention that you need to supply `--dirpath` due to having only documents enabled?
Also, if the message mentions the command line flag, shouldn't it error like in the CLI part instead of the lib function itself. When calling `openpype.lib.project_backpack.pack_project` manually there is no `--dirpath` argument to provide at all and the error message will still be unclear.

Reply:
```suggestion
            raise ValueError((
                "Destination dir must be passed for documents only pack."
                " Use '--dirpath {output dir path}' if using command line."
            ))
```

Reply:
> It says if using command line. so I would say that make it clear.

I meant more that I'd rather know what I failed to do when calling that function myself too. It gives no hint whatsoever.
We're providing CLI responses in a function that itself doesn't seem to be a CLI function. It's fine with me, but it's definitely confusing.

> Like this?

If this is intended to only ever be used through CLI then the whole reporting can be CLI focused: 
```suggestion
            raise ValueError((
                "Destination dir must be defined when using --dbonly."
                " Use '--dirpath {output dir path}' flag to specify directory."
            ))
```
That to me is much clearer.

If it isn't only CLI, then this should **not** report the CLI flags. I'd say the error reporting then is too mixed (or at the very least the argument names differ way to much `destination_dir` and `--dirpath` seem like different arguments due to the command line flag not mentioning _destination_).

I'd rather do the check directly in the CLI method to report the CLI friendly message, and in the function itself report it friendly for the function arguments.